powerauth-restful-security-spring-annotation/src/main/java/io/getlime/security/powerauth/rest/api/spring/provider/PowerAuthEncryptionProvider.java

```diff
@@ -79,4 +83,29 @@ public PowerAuthEncryptionProvider(PowerAuthClient powerAuthClient, HttpCustomiz
         }
     }
 
+    @Nonnull
+    @Override
+    public PowerAuthEciesEncryptorParameters getEciesEncryptorParameters(@Nullable String activationId, @Nonnull String applicationKey, @Nonnull String ephemeralPublicKey, @Nonnull String version, String nonce, Long timestamp) throws PowerAuthEncryptionException {
+        try {
+            final GetEciesDecryptorRequest eciesEncryptorRequest = new GetEciesDecryptorRequest();
+            eciesEncryptorRequest.setActivationId(activationId);
+            eciesEncryptorRequest.setApplicationKey(applicationKey);
+            eciesEncryptorRequest.setEphemeralPublicKey(ephemeralPublicKey);
+            eciesEncryptorRequest.setProtocolVersion(version);
+            eciesEncryptorRequest.setNonce(nonce);
+            eciesEncryptorRequest.setTimestamp(timestamp);
+            final GetEciesDecryptorResponse eciesEncryptorResponse = powerAuthClient.getEciesDecryptor(
+                    eciesEncryptorRequest,
+                    httpCustomizationService.getQueryParams(),
+                    httpCustomizationService.getHttpHeaders()
+            );
+
+            return new PowerAuthEciesEncryptorParameters(eciesEncryptorResponse.getSecretKey(), eciesEncryptorResponse.getSharedInfo2());
+        } catch (Exception ex) {
+            logger.warn("Get ECIES encryptor call failed, error: {}", ex.getMessage());
+            logger.debug(ex.getMessage(), ex);
+            throw new PowerAuthEncryptionException();
+        }
+    }
```

Comment:
This basically does the same thing as `getEciesDecryptorParameters`. In this context it's OK, but then in actual implementation it leads to a duplicit call to PAS.

Reply:
Ok, I will remove it.